components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OASParserUtil.java

```diff
@@ -1096,27 +1096,16 @@ public static void removePublisherSpecificInfo(Map<String, Object> extensions) {
         if (extensions == null) {
             return;
         }
-        if (extensions.containsKey(APIConstants.X_WSO2_AUTH_HEADER)) {
-            extensions.remove(APIConstants.X_WSO2_AUTH_HEADER);
-        }
-        if (extensions.containsKey(APIConstants.X_THROTTLING_TIER)) {
-            extensions.remove(APIConstants.X_THROTTLING_TIER);
-        }
-        if (extensions.containsKey(APIConstants.X_WSO2_CORS)) {
-            extensions.remove(APIConstants.X_WSO2_CORS);
-        }
-        if (extensions.containsKey(APIConstants.X_WSO2_PRODUCTION_ENDPOINTS)) {
-            extensions.remove(APIConstants.X_WSO2_PRODUCTION_ENDPOINTS);
-        }
-        if (extensions.containsKey(APIConstants.X_WSO2_SANDBOX_ENDPOINTS)) {
-            extensions.remove(APIConstants.X_WSO2_SANDBOX_ENDPOINTS);
-        }
-        if (extensions.containsKey(APIConstants.X_WSO2_BASEPATH)) {
-            extensions.remove(APIConstants.X_WSO2_BASEPATH);
-        }
-        if (extensions.containsKey(APIConstants.X_WSO2_TRANSPORTS)) {
-            extensions.remove(APIConstants.X_WSO2_TRANSPORTS);
-        }
+        extensions.remove(APIConstants.X_WSO2_AUTH_HEADER);
```

Comment:
can you please confirm extensions.remove() not throw NPE if the key is null. Looks like it does not throw NPE.

Reply:
if the element is not contained inside the array, it will not do anything. Refer https://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html#remove(java.lang.Object)